### ISSUE TITLE
Transcript effect name search: are arrays logical OR or AND?

### ISSUE THREAD

**Issue by david4096** [ga4gh/schemas#545]
When making a request for variation annotations an array of effect names can be sent:

    /**
    Only return variant annotations including these effects (SO terms).
    If null, return all variant annotations.
    */
    union { null, array<org.ga4gh.models.OntologyTerm> } effects = null;

https://github.com/ga4gh/schemas/blob/variation_annotation/src/main/resources/avro/alleleAnnotationmethods.avdl#L79

A single variation can have a number of transcript effects. Does one expect only annotations that have all of the specified effects are returned? Or are all variants with any of the requested effects returned?

In addition, transcript effects are part of the body of an annotation. When a search for a given effect is made but the variant has multiple transcript effects, are only the effects that match the search expected in the response? Or is it safe to assume all transcript effects shall be included with the annotation, regardless of the search terms?

**Comment by sarahhunt** [comment]
@david4096 - Only variant annotations with any of the specified effects should be returned. Only transcriptEffects with any of the specified effects should be returned. Does that make sense?

**Comment by david4096** [comment]
Yes, but please let me restate to be certain :).

Let's say we have two annotation records in our annotation set. The first (1) has one transcript effect named "downstream_gene_variant". The second (2) has two transcript effects, "downstream_gene_variant" and "splice_region_variant". If I perform a search of effect names with the array of ontology terms `["downstream_gene_variant", "splice_region_variant"]` I expect both annotation records (1) and (2) to be returned, with (2) including both of its transcript effects.

If I perform another search and request only `["downstream_gene_variant"]` I will be returned both (1) and (2), but (2) will only include the transcript effect that matches my search: the "splice_region_variant" transcript effects will not be included in the results.

Does that seem correct @sarahhunt ?

**Comment by diekhans** [comment]
@david4096  please restate by updating the documentation ;-)

**Comment by david4096** [comment]
I'm still not entirely clear that I've captured the desired functionality. Are the contents of the annotation record's transcript effects (which can contain arrays of ontology terms) filtered by the request?

Let's say our set has only one annotation and that annotation has two transcript effects T1, and T2. T1 and T2 have ontology effect terms [a, b] and [a, c], respectively:
```
    annotation
        T1: [a, b]
        T2: [a, c]
```
Now I perform a search using the effect term `[a]`. Do the results have the form:
```
    annotation
        T1: [a]
        T2: [a]
```
Or is the entirety of the annotation record from above returned?

Similarly if I request term `[b]` do the results have the form:
```
    annotation
        T1: [a, b]
```
or:
```
    annotation
        T1: [b]
```
or the entirety of the original record?

**Comment by sarahhunt** [comment]
Hi @david4096. Good question!

The full TranscriptEffect record should be returned, so using your examples above:

searching for [a] returns:    T1: [a, b]        T2: [a, c]
searching for [b] returns:    T1: [a, b] 

The filter removes variants (VariantAnnotations) and Transcripts (TranscriptEffects) which don't in combination have the effect of interest. Once you've identified a pairing of interest, the full description of the predicted effect is needed.

**Comment by diekhans** [comment]
Hi @sarahhunt , I am confused by this discussion. I don't see a search that returns TranscriptEffect .
I am worried that it would be error prone to return different version if a VariantAnnotation object with
the same id and different fields depending on the request.

**Comment by sarahhunt** [comment]
Hi @diekhans! VariantAnnotation groups TranscriptEffect's in a similar way that Variant groups calls. The effects filter on /variantannotations/search extracts a subset of TranscriptEffects in the same way that the 
callSetIds filter on /variants/search extracts a subset of Calls.

**Comment by diekhans** [comment]
Hi @sarahhunt

Doh.  I don't like that either; however, when in Rome ..

This can be re-considered when we look at the variant issues
and make them consistent.

**Comment by david4096** [comment]
Closed via #546
